src/averageAges.js

```diff
@@ -55,7 +67,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = !onlyWithSon
+    ? people
+      .filter(person => people.some(mother => mother.name === person.mother))
+    : people
+      .filter(person => person.sex.toLowerCase() === 'm'
+      && people.some(mother => mother.name === person.mother));
```

Comment:
You can use ternary operator inside filter callback to make your code a bit cleaner